modcc/printer/gpuprinter.cpp

```diff
@@ -445,7 +445,7 @@ void emit_api_body_cu(std::ostream& out, APIMethod* e, bool is_point_proc) {
     }
 
     if (!body->statements().empty()) {
-        out << "int tid_ = threadIdx.x + blockDim.x*blockIdx.x;\n";
+        cv_loop && out << "int tid_ = threadIdx.x + blockDim.x*blockIdx.x;\n";
```

Comment:
The generated kernel for `apply_events` requires the thread index, e.g
```C++
__global__
void apply_events(int mech_id_, mechanism_gpu_expsyn_pp_ params_, deliverable_event_stream_state events) {
    auto tid_ = threadIdx.x + blockDim.x*blockIdx.x;
    auto const ncell_ = events.n;

    if(tid_<ncell_) { /* .. */}
}  
```
~Are these being printed elsewhere?~

On further reading, I found where the `tid_` is calculated.

Should the `auto tid_ = ...` etc. be hoisted outside the `emit_api_body_cu`?

Reply:
Yes, I think that's a good idea. I just pushed the changes.